v2/dtos/responses/device.go

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package responses
+
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+)
+
+// AddDeviceResponse defines the Response Content for POST Device DTOs.
+// This object and its properties correspond to the AddDeviceResponse object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/AddDeviceResponse
+type AddDeviceResponse struct {
+	common.BaseWithIdResponse `json:",inline"`
+}
+
+// UpdateDeviceResponse defines the Response Content for PUT Device as pushed DTO.
+// This object and its properties correspond to the UpdateDeviceResponse object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/UpdateDeviceResponse
+type UpdateDeviceResponse struct {
+	common.BaseWithIdResponse `json:",inline"`
+}
+
+// DeviceResponse defines the Response Content for GET Device DTOs.
+// This object and its properties correspond to the DeviceResponse object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/DeviceResponse
+type DeviceResponse struct {
+	common.BaseResponse `json:",inline"`
+	device              dtos.Device
```

Comment:
Do we need the dtos.Device separate? Or just have the fields in-line like `DeviceProfileResponse` or does `DeviceProfileResponse` need to be reworked to use dto.DeviceProfile? 

Just need to be consistent and make any swagger updates to match what we do here.

Reply:
I modified the code to reuse the DTO struct in AddXXXRequest and XXXResponse.
For example:
```
type AddDeviceRequest struct {
	common.BaseRequest `json:",inline"`
	Device             dtos.Device `json:"device"`
}
type DeviceResponse struct {
	common.BaseResponse `json:",inline"`
	Device              dtos.Device `json:"device"`
}
```
And I will update the Swagger examples.